src/main/java/com/spotify/reaper/service/ClusterRepairScheduler.java

```diff
@@ -0,0 +1,143 @@
+    if (context.config.getAutoScheduling().hasScheduleSpreadPeriod()) {
+      return timeBeforeFirstSchedule.plus(scheduleIndex * context.config.getAutoScheduling().getScheduleSpreadPeriod().toMillis());
+    }
+    return timeBeforeFirstSchedule;
+  }
+
+  private void deleteRepairSchedule(Cluster cluster, String keyspace) {
+    Collection<RepairSchedule> scheduleCollection = context.storage.getRepairSchedulesForClusterAndKeyspace(cluster.getName(), keyspace);
+    scheduleCollection.forEach(repairSchedule -> {
+      context.storage.deleteRepairSchedule(repairSchedule.getId());
+      LOG.info("Scheduled repair deleted: {}", repairSchedule);
+    });
+  }
+
+  private boolean keyspaceCandidateForRepair(Cluster cluster, String keyspace) {
+    if (keyspace.toLowerCase().startsWith("system") || context.config.getAutoScheduling().getExcludedKeyspaces().contains(keyspace)) {
+      LOG.debug("Scheduled repair skipped for system keyspace {} in cluster {}.", keyspace, cluster.getName());
+      return false;
+    }
+    if (keyspaceHasNoTable(context, cluster, keyspace)) {
+      LOG.warn("No tables found for keyspace {} in cluster {}. No repair will be scheduled for this keyspace.", keyspace, cluster.getName());
+      return false;
+    }
+    return true;
+  }
+
+  private void createRepairSchedule(Cluster cluster, String keyspace, DateTime nextActivationTime) {
+    try {
+      RepairSchedule repairSchedule = CommonTools.storeNewRepairSchedule(
+          context,
+          cluster,
+          CommonTools.getNewOrExistingRepairUnit(context, cluster, keyspace, Collections.emptySet(), Boolean.FALSE),
+          context.config.getScheduleDaysBetween(),
+          nextActivationTime,
+          REPAIR_OWNER,
+          context.config.getSegmentCount(),
+          context.config.getRepairParallelism(),
+          context.config.getRepairIntensity()
+      );
+      LOG.info("Scheduled repair created: {}", repairSchedule);
+    } catch (ReaperException e) {
+      Throwables.propagate(e);
+    }
+  }
+
+  private boolean keyspaceHasNoTable(AppContext context, Cluster cluster, String keyspace)  {
+    try (JmxProxy jmxProxy = context.jmxConnectionFactory.connectAny(cluster)) {
+      Set<String> tables = jmxProxy.getTableNamesForKeyspace(keyspace);
+      return tables.isEmpty();
+    } catch (ReaperException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  private static class ScheduledRepairDiffView {
+    private final ImmutableSet<String> keyspacesThatRequireSchedules;
+    private final ImmutableSet<String> keyspacesDeleted;
+
+    public static ScheduledRepairDiffView compareWithExistingSchedules(AppContext context, Cluster cluster) throws ReaperException {
+      return new ScheduledRepairDiffView(context, cluster);
+    }
+
+    public ScheduledRepairDiffView(AppContext context, Cluster cluster) throws ReaperException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)